### PR TITLE
use #attach method insted of #embed to get rid of deprecation warning.

### PR DESCRIPTION
**Pull request by pruan-rht**
This addresses [OCPQE-3985](https://issues.redhat.com/browse/OCPQE-3985) @yapei @liangxia PTAL.

### CODE REVIEW

File: lib/log.rb [hunk 1]
```diff
@@ -158,9 +158,8 @@ def print(msg)
 
     # supports embedding content similar with same semantics as Cucumber
     def embed(src, mime_type, label)
-      if self.class.runtime.respond_to? :embed
-        info "embedding #{label.inspect}"
-        self.class.runtime.embed src, mime_type, label
+      if self.class.runtime.respond_to? :attach
+        self.class.runtime.attach src, mime_type
```
Comment (liangxia) [inline lib/log.rb:162]: @pruan-rht Thank you for the fix.
@yapei  Could you help to test the changes ? Thanks.

Comment (pruan-rht) [reply lib/log.rb:162]: @liangxia I tested locally by using a bad xyaml...the screen capture took place and the html is saved.

**Comment by yapei** [comment]
I tested and found some issues, copying my messages in Slack here
```
There are some issues:
We still need this line info "embedding #{label.inspect}"  so that we can find the correct screenshot for specific failure when more than 1 screenshot is saved
The screenshot is saved with name test case execution attachment  and text  format where we .png  file is expected
```

**Comment by pruan-rht** [comment]
@yapei I tried to run the `embed` methods as what the current state of github repo and the screen shot is just one big one.  Perhaps this is related to cuke5.  Also, I checked in a patch to get rid of the warning.

**Comment by yapei** [comment]
> @yapei I tried to run the `embed` methods as what the current state of github repo and the screen shot is just one big one. Perhaps this is related to cuke5. Also, I checked in a patch to get rid of the warning.
`the screen shot is just one big one`  yes, is it saved as name `test case execution attachment` which should be `2021-xx-xxT07:00:40+00:00-screenshot.png`

**Comment by pruan-rht** [comment]
superseded by #2025, will open another one for the browser warnings.

**Pull request closed without merging**